spec/solver_spec.rb

```diff
@@ -0,0 +1,50 @@
+require_relative '../solver'
+
+describe Solver do
+  describe '#factorial' do
+    it 'accepts n as a single' do
+      solver = Solver.new
+      expect(solver.factorial(1)).to eq 1
+    end
+
+    it 'returns 1 when given 0' do
+      solver = Solver.new
+      expect(solver.factorial(0)).to eq 1
+    end
+
+    it 'returns 400 when given 20' do
+      solver = Solver.new
+      expect(solver.factorial(3)).to eq 6
+    end
+
+    it 'returns an error when given a negative number' do
+      solver = Solver.new
+      expect { solver.factorial(-1) }.to raise_error ArgumentError
+    end
+  end
+
+  describe '#reverse' do
+    it 'returns a reverse word' do
+      solver = Solver.new
+      expect(solver.reverse('hello')).to eq 'olleh'
+    end
+  end
+  describe '#fizzbuzz' do
+    it 'returns fizzbuzz when given 15' do
+      solver = Solver.new
+      expect(solver.fizzbuzz(15)).to eq 'fizzbuzz'
+    end
+    it 'returns fizz when given 3' do
+      solver = Solver.new
+      expect(solver.fizzbuzz(3)).to eq 'fizz'
+    end
+    it 'returns buzz when given 5' do
+      solver = Solver.new
+      expect(solver.fizzbuzz(5)).to eq 'buzz'
+    end
+    it 'returns 1 when given 1' do
+      solver = Solver.new
+      expect(solver.fizzbuzz(1)).to eq '1'
+    end
```

Comment:
- You have done great so far :+1: But there are a few changes you need to make before moving to the next milestone. :smile: :+1: 

- - Screenshot reference:
![Selection_036](https://github.com/siddghosh108/TDD-Project/assets/107632780/73f9ae8d-d52a-4abf-b70e-258336d4ab3f)

- [ ] You did great implementing the tests for the `fizzbuzz`.  I've taken a close look at the test cases for the `fizzbuzz` method, and I appreciate the effort you've put into it. :+1: :100: 
However, it seems there's a slight misinterpretation in the expected behavior. The requirement for the `fizzbuzz` method is to return "fizz" when the given number is divisible by 3, and  `'buzz' when N is divisible by 5 .. etc. Currently, the test case for divisibility by 3 is checking against the number 3 itself.

To align with the specification, please update the test case to check for a number that is divisible by 3. Here's an example:

```ruby
it 'returns fizz when given a number divisible by 3' do
  solver = Solver.new
  expect(solver.fizzbuzz(6)).to eq 'fizz'
end
```
You need to also update the `it bock` of the other test cases to better articulate the requirement i.e:  `returns "buzz" When N is divisible by 5`, `returns "fizzbuzz" When N is divisible by 5`, `returns N as a string when N is not divisible by 3 and 5`

Reply:
I have made the exact changes you asked for, please see the below code:
describe '#fizzbuzz' do
    solver = Solver.new
    it 'returns fizzbuzz when given a number divisible by both 3 and 5' do
      expect(solver.fizzbuzz(15)).to eq 'fizzbuzz'
    end
    it 'returns fizz when given a number divisible by 3' do
      expect(solver.fizzbuzz(6)).to eq 'fizz'
    end
    it 'returns buzz when given a number divisible by 5' do
      expect(solver.fizzbuzz(5)).to eq 'buzz'
    end
    it 'returns the number as a string when given a number not divisible by 3 and 5' do
      expect(solver.fizzbuzz(1)).to eq '1'
    end
  end